server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -449,6 +449,26 @@ public void consumerExists(
         }
     }
 
+    @ApiOperation(
+        notes = "Checks for the existence of a Consumer in bulk. This API returns uuids of non-existing" +
+        "consumer",
+        value = "")
+    @ApiResponses({ @ApiResponse(code = 400, message = "Return 404 when empty or null array is passed"),
+        @ApiResponse(code = 200, message = "Return array of UUIDs of consumers that doesn't exist or " +
+        "cannot be accessed") })
+    @POST
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("exists")
+    public List<String> consumerExistsBulk(ArrayList<String> consumerUuid) throws BadRequestException {
```

Comment:
Does this absolutely need to be an ArrayList, or will any list (or collection/iterable) work here?

Also, if it's a collection, the variable name should reflect that with a plural.

Reply:
Fixing it. Using a Generic collection representation for I/O.